### PR TITLE
Cutting unnecessary dependencies for NRP and policy packages build on Ubuntu 14 workflow

### PR DESCRIPTION
**Pull request by MariusNi**
## Description

This PR cuts unnecessary dependencies for NRP and policy packages build on Ubuntu 14 workflow. As fewer dependencies we have to install there, as safer we are on such an old distribution. With this occasion also adjusting some of the names (as the workflow is for building 'OSConfig for MC' aka NRP and policy packages, not publishing it), cutting unnecessary steps (building CMake 3, signing off the policy package, etc.), enabling for PRs and nightly runs the new sanitizers CI, and also used to run our full CI validation on the new policy packages stored at https://github.com/Azure/azure-osconfig/releases/tag/test_policy_package.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: .github/workflows/nrp-build.yml [hunk 1]
```diff
@@ -41,14 +41,5 @@ jobs:
       artifact: ${{ inputs.target }}-${{ inputs.arch }}
       package-type: ${{ inputs.package-type }}
       container-tag: ${{ inputs.container-tag }}
-      buildForMachineConfiguration: ${{ inputs.buildForMachineConfiguration }}
-      release: true
-
-  sign:
```
Comment (robertschaedler3) [inline .github/workflows/nrp-build.yml:47]: Should the NRP be signed?

File: devops/docker/ubuntu-14.04-amd64/Dockerfile [hunk 2]
```diff
@@ -2,34 +2,11 @@ FROM ubuntu:trusty
...
     file
 
-WORKDIR /git
-
-# CMake
-RUN git clone https://github.com/Kitware/CMake --recursive -b v3.21.7
-RUN cd CMake && ./bootstrap && make -j$(nproc) && make install && hash -r && rm -rf /git/CMake
-
-# GTest
-RUN git clone https://github.com/google/googletest --recursive -b release-1.10.0
```
Comment (robertschaedler3) [inline devops/docker/ubuntu-14.04-amd64/Dockerfile:30]: Are unit tests run on Ubuntu 14?

File: devops/docker/ubuntu-14.04-amd64/Dockerfile [hunk 3]
```diff
@@ -2,34 +2,11 @@ FROM ubuntu:trusty
...
-    wget \
-    gcovr\
-    jq \
-    bc \
     file
 
-WORKDIR /git
-
-# CMake
-RUN git clone https://github.com/Kitware/CMake --recursive -b v3.21.7
```
Comment (robertschaedler3) [inline devops/docker/ubuntu-14.04-amd64/Dockerfile:26]: Do we need CMake to build on Ubuntu 14?

**Review by robertschaedler3 — APPROVED** [review]

File: .github/workflows/nrp-build.yml (hunk 1 above)
Comment (MariusNi) [reply .github/workflows/nrp-build.yml:47]: No.

File: devops/docker/ubuntu-14.04-amd64/Dockerfile (hunk 2 above)
Comment (MariusNi) [reply devops/docker/ubuntu-14.04-amd64/Dockerfile:30]: None

File: devops/docker/ubuntu-14.04-amd64/Dockerfile (hunk 3 above)
Comment (MariusNi) [reply devops/docker/ubuntu-14.04-amd64/Dockerfile:26]: Yes, but the 'inbox' 2.x one is enough, no need for v 3.

Comment (robertschaedler3) [reply devops/docker/ubuntu-14.04-amd64/Dockerfile:26]: Ahh I see, thanks for the clarification  👍

File: .github/workflows/github.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-name: Publish (GitHub)
+name: Build 'OSConfig for MC' (NRP and policy packages)
```
Comment (AhmedBM) [inline .github/workflows/github.yml:1]: This is a publishing worklow, sure it includes building it but the root name should be `Publish ...` something. Maybe `Publish 'OSConfig for MC (NRP and policy packages)` is a better name?

Comment (MariusNi) [reply .github/workflows/github.yml:1]: It appears to work fine as it is and new name better reflects the actual purpose.

Comment (MariusNi) [reply .github/workflows/github.yml:1]: As discussed, we'll change prefix to 'Publish' when/if we'll make this workflow also publish the package (e.g. to a release)

**Review by AhmedBM — APPROVED** [review]

File: .github/workflows/nrp-build.yml (hunk 1 above)
Comment (simonjaeger) [reply .github/workflows/nrp-build.yml:47]: Why not?

Comment (MariusNi) [reply .github/workflows/nrp-build.yml:47]: Not a signed off or even a proposed requirement.

**Pull request merged**